webui/react/src/stores/userRoles.tsx

```diff
@@ -1,31 +1,31 @@
-import React, { createContext, PropsWithChildren, useCallback, useContext, useState } from 'react';
+import { observable, WritableObservable } from 'micro-observables';
+import React, { createContext, PropsWithChildren, useCallback, useContext, useRef } from 'react';
 
 import { getPermissionsSummary } from 'services/api';
+import { isEqual } from 'shared/utils/data';
 import { UserAssignment, UserRole } from 'types';
 import handleError from 'utils/error';
 import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+import { useValueMemoizedObservable } from 'utils/observable';
 
 type UserRolesAndAssignmentsContext = {
-  updateUserAssignments: (
-    fn: (r: Loadable<UserAssignment[]>) => Loadable<UserAssignment[]>,
-  ) => void;
-  updateUserRoles: (fn: (r: Loadable<UserRole[]>) => Loadable<UserRole[]>) => void;
-  userAssignments: Loadable<UserAssignment[]>;
-  userRoles: Loadable<UserRole[]>;
+  userAssignments: WritableObservable<Loadable<UserAssignment[]>>;
+  userRoles: WritableObservable<Loadable<UserRole[]>>;
```

Comment:
We may want to keep these private

Reply:
that's a good idea
I think I've been basing it on the wrong changes, I should do a revision to make it more like the updated https://github.com/determined-ai/determined/pull/5817

Reply:
OK, I made these private, set it up as a static class, and fixed a unit test issue